Sources/AuthFoundation/JWT/Internal/DefaultIDTokenValidator.swift

```diff
@@ -18,61 +18,72 @@ import CommonCrypto
+    var checks: [ValidationCheck] = ValidationCheck.allCases
+    
+    enum ValidationCheck: CaseIterable {
+        case issuer, audience, scheme, algorithm, expirationTime, issuedAtTime, nonce, maxAge, subject
+    }
     
     func validate(token: JWT, issuer: URL, clientId: String, context: IDTokenValidatorContext?) throws {
         guard let tokenIssuerString = token.issuer,
-              let tokenIssuer = URL(string: tokenIssuerString)
-        else {
-            throw JWTError.invalidIssuer
-        }
-
-        guard tokenIssuer.absoluteString == issuer.absoluteString
+              let tokenIssuer = URL(string: tokenIssuerString),
+              tokenIssuer.absoluteString == issuer.absoluteString
         else {
             throw JWTError.invalidIssuer
         }
-
-        guard token[.audience] == clientId else {
-            throw JWTError.invalidAudience
-        }
-
-        guard tokenIssuer.scheme == "https" else {
-            throw JWTError.issuerRequiresHTTPS
-        }
-        
-        guard token.header.algorithm == .rs256 else {
-            throw JWTError.unsupportedAlgorithm(token.header.algorithm)
-        }
         
-        guard let expirationTime = token.expirationTime,
-              expirationTime > Date.nowCoordinated else {
-            throw JWTError.expired
-        }
-        
-        guard let issuedAt = token.issuedAt,
-              abs(issuedAt.timeIntervalSince(Date.nowCoordinated)) <= issuedAtGraceInterval
-        else {
-            throw JWTError.issuedAtTimeExceedsGraceInterval
-        }
-        
-        guard token["nonce"] == context?.nonce else {
-            throw JWTError.nonceMismatch
-        }
-
-        if let maxAge = context?.maxAge {
-            guard let authTime = token.authTime else {
-                throw JWTError.invalidAuthenticationTime
-            }
-            
-            let elapsedTime = issuedAt.timeIntervalSince(authTime)
-            guard elapsedTime > 0 && elapsedTime <= maxAge else {
-                throw JWTError.exceedsMaxAge
+        for check in checks {
+            switch check {
+            case .issuer:
+                guard tokenIssuer.absoluteString == issuer.absoluteString
+                else {
+                    throw JWTError.invalidIssuer
+                }
+            case .audience:
+                guard token[.audience] == clientId else {
+                    throw JWTError.invalidAudience
+                }
+            case .scheme:
+                guard tokenIssuer.scheme == "https" else {
+                    throw JWTError.issuerRequiresHTTPS
+                }
+            case .algorithm:
+                guard token.header.algorithm == .rs256 else {
+                    throw JWTError.unsupportedAlgorithm(token.header.algorithm)
+                }
+            case .expirationTime:
+                guard let expirationTime = token.expirationTime,
+                      expirationTime > Date.nowCoordinated else {
```

Comment:
Can you put this `else` on a new line?

The style guide I've been working with is inline `guard else` can be fine for single expressions, but for multiple expressions, we put the `else` on its own line.  This is because otherwise the expression, and the code, can be hard to distinguish from one another.

Reply:
The sounds good. Maybe we can add this to the linter, so its not something we manually check for down the line. I'll look into it later.